perun-core/src/main/java/cz/metacentrum/perun/core/impl/GroupsManagerImpl.java

```diff
@@ -1027,4 +1027,34 @@ public MemberGroupStatus getTotalMemberGroupStatus(PerunSession session, Member
 			throw new InternalErrorException(e);
 		}
 	}
+
+	@Override
+	public List<Group> getGroupsForAutoRegistration(PerunSession sess, Vo vo) {
+		try {
+			return jdbc.query("select " + groupMappingSelectQuery + " from groups join groups_to_register on groups.id=groups_to_register.group_id " +
+				"where groups_to_register.vo_id=?", GROUP_MAPPER, vo.getId());
```

Comment:
I don't think we need to save the voId in the `group_to_register` table. You can then modify the SQL something like this:
```suggestion
			return jdbc.query("select " + groupMappingSelectQuery + " from groups where vo_id=? and id IN (select group_id from groups_to_register)", GROUP_MAPPER, vo.getId());
```

Reply:
Fixed.